installers/agent/release/agent.sh

```diff
@@ -101,15 +98,18 @@ AGENT_STARTUP_ARGS="-Dcruise.console.publish.interval=10 -Xms$AGENT_MEM -Xmx$AGE
 if [ "$TMPDIR" != "" ]; then
     AGENT_STARTUP_ARGS="$AGENT_STARTUP_ARGS -Djava.io.tmpdir=$TMPDIR"
 fi
+if [ "$USE_URANDOM" != "false" ] && [ -e "/dev/urandom" ]; then
+    AGENT_STARTUP_ARGS="$AGENT_STARTUP_ARGS -Djava.security.egd=file:/dev/./urandom"
```

Comment:
No. It doesn't look like it is fixed yet.

```
$ cat JRand.java
import java.security.SecureRandom;
class JRand {
  public static void main(String args[]) throws Exception {
    System.out.println("Ok: " +
      SecureRandom.getInstance("SHA1PRNG").nextLong());
  }
}

$ java -version
java version "1.7.0_79"
OpenJDK Runtime Environment (IcedTea 2.5.6) (7u79-2.5.6-0ubuntu1.14.04.1)
OpenJDK 64-Bit Server VM (build 24.79-b02, mixed mode)

$ time java -Djava.security.egd=file:/dev/./urandom JRand
Ok: -4364682451133414604

real    0m0.129s
user    0m0.120s
sys     0m0.020s

$ time java -Djava.security.egd=file:/dev/./urandom JRand
Ok: -9054433491496715268

real    0m0.130s
user    0m0.130s
sys     0m0.010s

$ time java -Djava.security.egd=file:/dev/./urandom JRand
Ok: 7166545031684433590

real    0m0.125s
user    0m0.130s
sys     0m0.020s

$ time java JRand
[STUCK]
```

Did a `kill -3` and found this in the stack trace:

```
"main" prio=10 tid=0x00007feef000a000 nid=0x44 runnable [0x00007feef9141000]
   java.lang.Thread.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(FileInputStream.java:272)
        at sun.security.provider.SeedGenerator$URLSeedGenerator.getSeedBytes(SeedGenerator.java:551)
        at sun.security.provider.SeedGenerator.generateSeed(SeedGenerator.java:139)
        at sun.security.provider.SecureRandom$SeederHolder.<clinit>(SecureRandom.java:197)
        at sun.security.provider.SecureRandom.engineNextBytes(SecureRandom.java:214)
        - locked <0x00000000f229f3d8> (a sun.security.provider.SecureRandom)
        at java.security.SecureRandom.nextBytes(SecureRandom.java:455)
        - locked <0x00000000f229fcc8> (a java.security.SecureRandom)
        at java.security.SecureRandom.next(SecureRandom.java:477)
        at java.util.Random.nextLong(Random.java:334)
        at JRand.main(JRand.java:4)
```

Then, killed that and tried this:

```
$ time java -Djava.security.egd=file:/dev/urandom JRand
```

Stuck again and still the same stack trace:

```
"main" prio=10 tid=0x00007f518400a000 nid=0xa6 runnable [0x00007f518daca000]
   java.lang.Thread.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(FileInputStream.java:272)
        at sun.security.provider.SeedGenerator$URLSeedGenerator.getSeedBytes(SeedGenerator.java:551)
        at sun.security.provider.SeedGenerator.generateSeed(SeedGenerator.java:139)
        at sun.security.provider.SecureRandom$SeederHolder.<clinit>(SecureRandom.java:197)
        at sun.security.provider.SecureRandom.engineNextBytes(SecureRandom.java:214)
        - locked <0x00000000f229f3d8> (a sun.security.provider.SecureRandom)
        at java.security.SecureRandom.nextBytes(SecureRandom.java:455)
        - locked <0x00000000f229fcc8> (a java.security.SecureRandom)
        at java.security.SecureRandom.next(SecureRandom.java:477)
        at java.util.Random.nextLong(Random.java:334)
        at JRand.main(JRand.java:4)
```

`/dev/./urandom` still works fine:

```
$ time java -Djava.security.egd=file:/dev/./urandom JRand
Ok: 338890852372789233

real    0m0.191s
user    0m0.120s
sys     0m0.020s

$ time java -Djava.security.egd=file:/dev/./urandom JRand
Ok: -1104126029115977571

real    0m0.122s
user    0m0.120s
sys     0m0.010s

$ time java -Djava.security.egd=file:/dev/./urandom JRand
Ok: 5840685795943611680

real    0m0.127s
user    0m0.130s
sys     0m0.010s
```

All of this was tried on Centos running on docker. strace shows both files are opened, in both cases, though. Probably, in the /dev/./urandom case, it doesn't try reading from it.

```
$ strace -f -e file java -Djava.security.egd=file:/dev/urandom JRand 2>&1 | grep random
[pid   156] stat("/dev/random", {st_mode=S_IFCHR|0666, st_rdev=makedev(1, 8), ...}) = 0
[pid   156] stat("/dev/urandom", {st_mode=S_IFCHR|0666, st_rdev=makedev(1, 9), ...}) = 0
[pid   156] open("/dev/random", O_RDONLY) = 12
[pid   156] open("/dev/urandom", O_RDONLY) = 13
[pid   156] open("/dev/random", O_RDONLY) = 14
^C
[WAS STUCK]

$ strace -f -e file java -Djava.security.egd=file:/dev/./urandom JRand 2>&1 | grep random
[pid   181] stat("/dev/random", {st_mode=S_IFCHR|0666, st_rdev=makedev(1, 8), ...}) = 0
[pid   181] stat("/dev/urandom", {st_mode=S_IFCHR|0666, st_rdev=makedev(1, 9), ...}) = 0
[pid   181] open("/dev/random", O_RDONLY) = 12
[pid   181] open("/dev/urandom", O_RDONLY) = 13
[pid   181] open("/dev/./urandom", O_RDONLY) = 14
```